File: taskmanagement-ui/src/features/activity/api/ActivityLogService.ts
import { apiClient as api } from '@/api/axios';
import { API_ENDPOINTS } from '@/utils/constants';
import type { ApiResponse } from '@/features/auth/types/auth';
import type { PageResponse } from '@/features/users/types/user';

export interface ActivityLogResponse {
  id: number;
  action: string;
  description: string;
  username: string;
  taskTitle: string;
  module: string;
  entityId: number;
  ipAddress?: string;
  result?: string;
  createdAt: string;
}

export interface ActivityLogSearchParams {
  username?: string;
  module?: string;
  action?: string;
  result?: string;
  ipAddress?: string;
  startDate?: string;
  endDate?: string;
  page?: number;
  size?: number;
}

export const ActivityLogService = {
  searchLogs: async (params: ActivityLogSearchParams): Promise<PageResponse<ActivityLogResponse>> => {
    const response = await api.get<ApiResponse<PageResponse<ActivityLogResponse>>>(API_ENDPOINTS.ACTIVITY_LOGS_SEARCH, { params });
    return response.data.data;
  }
};

File: src/main/java/com/phong/taskmanagement/service/interfaces/ActivityLogService.java
package com.phong.taskmanagement.service.interfaces;

import java.util.List;

import com.phong.taskmanagement.dto.request.CreateActivityLogRequest;
import com.phong.taskmanagement.dto.response.ActivityLogResponse;

public interface ActivityLogService {

    ActivityLogResponse createLog(
            CreateActivityLogRequest request
    );

    List<ActivityLogResponse> getAllLogs();

    org.springframework.data.domain.Page<ActivityLogResponse> searchLogs(
            String username,
            String module,
            String action,
            String result,
            String ipAddress,
            String startDate,
            String endDate,
            org.springframework.data.domain.Pageable pageable
    );

    ActivityLogResponse log(
            Long userId,
            Long entityId,
            String module,
            String action,
            String description
    );

    // Keep backwards compatible version for existing code
    ActivityLogResponse log(
            Long userId,
            Long taskId,
            String action,
            String description
    );
}
